### ISSUE TITLE
Resource `args` is always the same as `previous.args`

### ISSUE THREAD

**Issue by nag5000** [NullVoxPopuli/ember-resources#195]
When `previous` is not null, its `args` are equal to the current `args`. It is a little unexpected, is it a bug?

Reproduction: https://stackblitz.com/edit/github-zip5hx?file=app/controllers/application.js

```js
class MyResource extends Resource {
  constructor(owner, args, previous) {
    super(owner, args, previous);
    const { filter } = args.named;
    const { filter: oldFilter } = previous ? previous.args.named : {};
    this.value = { filter, oldFilter };
  }
}

export default class ApplicationController extends Controller {
  @tracked filter = '';

  res = useResource(this, MyResource, () => ({ filter: this.filter }));

  @action
  setFilter() {
    const filter = window.prompt('set filter', this.filter);
    this.filter = filter;
  }
}
```

```handlebars
filter: {{this.res.value.filter}}
<br>oldFilter: {{this.res.value.oldFilter}}
<br><button {{on "click" this.setFilter}}>Set filter</button>
```

1. Click "Set filter" and enter "foobar"
2. `filter: foobar <br> oldFilter: foobar` will be rendered instead of `filter: foobar <br> oldFilter: <empty string>`.

**Comment by NullVoxPopuli** [comment]
I've reproduced this in an automated test and will get a fix out. thank you for reporting!

**Comment by NullVoxPopuli** [comment]
this might be a bug in ember :thinking: ?

**Comment by NullVoxPopuli** [comment]
still investigating -- stay tuned!

**Comment by NullVoxPopuli** [comment]
fixed in v4.5.0, lemme know what you think :)

(have to use the new base class and `modify` hook)
